packages/plugin-exec/sources/ExecFetcher.ts

```diff
@@ -1,9 +1,40 @@
-import {Fetcher, FetchOptions, MinimalFetchOptions}                    from '@yarnpkg/core';
-import {Locator, MessageName}                                          from '@yarnpkg/core';
-import {execUtils, scriptUtils, structUtils, tgzUtils}                 from '@yarnpkg/core';
-import {Filename, NodeFS, PortablePath, npath, ppath, toFilename, xfs} from '@yarnpkg/fslib';
-
-import {PROTOCOL}                                                      from './constants';
+import {execUtils, scriptUtils, structUtils, tgzUtils}                     from '@yarnpkg/core';
+import {Locator, MessageName}                                              from '@yarnpkg/core';
+import {Fetcher, FetchOptions, MinimalFetchOptions}                        from '@yarnpkg/core';
+import {Filename, PortablePath, npath, ppath, toFilename, xfs, NativePath} from '@yarnpkg/fslib';
+
+import {PROTOCOL}                                                          from './constants';
+import {getGeneratorPath}                                                  from './execUtils';
+
+/**
+ * Contains various useful details about the execution context.
+ */
+export interface ExecEnv {
+  /**
+   * The absolute path of the temporary directory where the script runs. Equal to `process.cwd()`.
+   */
+  tempDir: NativePath;
+  /**
+   * The Locator identifying the generator package.
+   */
+  locator: Locator;
+  /**
+   * The absolute path of the generator file. Equal to `process.argv[1]`.
+   */
+  generatorPath: NativePath;
```

Comment:
Is this useful? What's the difference with `__filename`?

Reply:
There's no difference with `__filename`. Now it's also useless, since we changed the `generatorPath` to a temp directory.